niceman/resource/tests/test_ssh.py

```diff
@@ -23,51 +23,57 @@
 from nose import SkipTest
 
 @pytest.fixture(scope='module')
-def setup_docker():
+def setup_ssh():
     """pytest fixture for tests needing a running docker container
 
     on setup, this fixture ensures that a docker container that maps 
-    host port 49000 to container port 22 is running and starts one if necessary
+    host port 49000 to container port 22 is running and starts one if necessary.
+
+    Fixture yields parameters of the ssh connection suitable to be used as is
+    for ssh type resource.
 
     on teardown, this fixture stops the docker container if it was started by 
     the fixture
     """
+
     skip_if_no_network()
+    skip_ssh()
     stdout, _ = Runner().run(['docker', 'ps'])
     if '0.0.0.0:49000->22/tcp' in stdout:
         stop_container = False
     else:
+        params = {
+            'host': 'localhost',
+            'user': 'root',
+            'password': 'root',
+            'port': '49000',
+        }
         args = ['docker', 
                 'run', 
                 '-d', 
                 '--rm', 
                 '-p', 
-                '49000:22', 
+                '{port}:22'.format(**params),
                 'rastasheep/ubuntu-sshd:14.04']
         stdout, _ = Runner().run(args)
         container_id = stdout.strip()
         stop_container = True
-    yield
+    yield params
```

Comment:
params is not defined if a suitable container is already running (if the test on line 42 succeeds).  You might also want to use params['port'] in this test so 49000 only appears once.

Reply:
ho ho... thanks! fixing up!